src/models/nulog/NuLogParser.py

```diff
@@ -0,0 +1,516 @@
+        self.df_log = None
+        self.log_format = log_format
+        self.tokenizer = LogTokenizer(filters)
+
+        if not os.path.exists(self.savePath):
+            os.makedirs(self.savePath)
+        self.model_path = os.path.join(self.savePath, self.logName + ".pt")
+
+    def num_there(self, s):
+        digits = [i.isdigit() for i in s]
+        return True if np.mean(digits) > 0.0 else False
+
+    def save_model(self, model, model_opt, epoch, loss):
+        torch.save(
+            {
+                "epoch": epoch,
+                "model_state_dict": model.state_dict(),
+                "optimizer_state_dict": model_opt.state_dict(),
+                "loss": loss,
+            },
+            self.model_path,
+        )
+
+    def load_model(self, model, model_opt):
+        if using_GPU:
+            ckpt = torch.load(self.model_path)
+        else:
+            ckpt = torch.load(self.model_path, map_location=torch.device("cpu"))
+        try:
+            model_opt.load_state_dict(ckpt["optimizer_state_dict"])
+            model.load_state_dict(ckpt["model_state_dict"])
+            epoch = ckpt["epoch"]
+            loss = ckpt["loss"]
+            return epoch, loss
+        except:  ## TODO: remove this try except when we use the new save function.
+            logging.error("loading trained model with old format.")
+            model.load_state_dict(ckpt)
+
+    def train(
+        self,
+        data_tokenized,
+        batch_size=32,
+        mask_percentage=1.0,
+        pad_len=64,
+        N=1,
+        d_model=256,
+        dropout=0.1,
+        lr=0.001,
+        betas=(0.9, 0.999),
+        weight_decay=0.005,
+        nr_epochs=5,
+        num_samples=0,
+        step_size=100,
+    ):
+        self.mask_percentage = mask_percentage
+        self.pad_len = pad_len
+        self.batch_size = batch_size
+        self.N = N
+        self.d_model = d_model
+        self.dropout = dropout
+        self.lr = lr
+        self.betas = betas
+        self.weight_decay = weight_decay
+        self.num_samples = num_samples
+        self.nr_epochs = nr_epochs
+        self.step_size = step_size
+
+        logging.debug("learning rate : " + str(self.lr))
+        # test!
+        # data_tokenized = data_tokenized[:10]
+        transform_to_tensor = transforms.Lambda(lambda lst: torch.tensor(lst))
+
+        criterion = nn.CrossEntropyLoss()
+        model = self.make_model(
+            self.tokenizer.n_words,
+            self.tokenizer.n_words,
+            N=self.N,
+            d_model=self.d_model,
+            d_ff=self.d_model,
+            dropout=self.dropout,
+            max_len=self.pad_len,
+        )
+
+        if using_GPU:
+            model.cuda()
+        model_opt = torch.optim.Adam(
+            model.parameters(),
+            lr=self.lr,
+            betas=self.betas,
+            weight_decay=self.weight_decay,
+        )
+
+        if (self.logName + ".pt") in os.listdir(self.savePath):
```

Comment:
rather than remove it, I removed logName and have 
`model_name = "nulog_model_latest.pt"`  instead.
also I have `self.savePath = "output/"` rather than delete it. 
I think it's better to put output model files and vocabs in a folder.

Reply:
fix updated